x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java

```diff
@@ -497,8 +489,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) throws IOException {
         boolean licenseSpecMode = params.paramAsBoolean(LICENSE_SPEC_VIEW_MODE, false);
         boolean restViewMode = params.paramAsBoolean(REST_VIEW_MODE, false);
-        boolean hideEnterprise = params.paramAsBoolean(XCONTENT_HIDE_ENTERPRISE, false);
-
         boolean previouslyHumanReadable = builder.humanReadable();
```

Comment:
The only way this could have ever been true is via REST API compatibility with v7: https://github.com/elastic/elasticsearch/blob/8.x/x-pack/plugin/core/src/main/java/org/elasticsearch/license/RestGetLicenseAction.java#L66-L72 . 9.x does not support REST API compatibility with v7, so this value will always be false. (If you used v7 compatibility, you could set the `accept_enterprise` parameter to `false` which resulted in `XCONTENT_HIDE_ENTERPRISE` being `true`, now that v7 compatibility is gone there is no way to set `accept_enterprise` to `false` (via other validation) , so `XCONTENT_HIDE_ENTERPRISE` is always false) This change is adjusting the code for `hideEnterprise` is always false.